### PR TITLE
Refactoring basepage code

### PR DESCRIPTION
**Pull request by raghavgarg1996**
Optimizing the basepage code - removing nested if else

**ReportPortal**

**Web**
http://reportportal.jio.com/ui/#qecc/launches/all/255526
http://reportportal.jio.com/ui/#qecc/launches/all/255539

**App**
http://reportportal.jio.com/ui/#qecc/launches/all/255543
http://reportportal.jio.com/ui/#qecc/launches/all/255549

**Applitools**

**Web**
https://jioeyes.applitools.com/app/test-results/00000251705208045628/?accountId=8euMeI_LvUiaNtUWS_HwsA__
https://jioeyes.applitools.com/app/test-results/00000251705207323493/?accountId=8euMeI_LvUiaNtUWS_HwsA__

**App**
https://jioeyes.applitools.com/app/test-results/00000251705207088293/?accountId=8euMeI_LvUiaNtUWS_HwsA__
https://jioeyes.applitools.com/app/test-results/00000251705206907430/?accountId=8euMeI_LvUiaNtUWS_HwsA__

### CODE REVIEW

File: src/main/java/com/znsio/rpap/pages/BasePage.java [hunk 1]
```diff
@@ -9,51 +9,51 @@
...
-            Select dropdown = new Select(ele);
-            dropdown.selectByVisibleText(dataToInput);
-
-        } else if (tagName.equals("textarea")) {
-
-            scrollToView(by);
-            ele.clear();
-            ele.sendKeys(new CharSequence[]{dataToInput});
-            ele.sendKeys(new CharSequence[]{Keys.TAB});
+    private void handleSelectInput(WebElement element, String dataToInput) {
```
Comment (anandbagmar) [inline src/main/java/com/znsio/rpap/pages/BasePage.java:37]: what does "handle" really do?

File: src/main/java/com/znsio/rpap/pages/BasePage.java [hunk 2]
```diff
@@ -9,51 +9,51 @@
...
+        Select dropdown = new Select(element);
+        dropdown.selectByVisibleText(dataToInput);
+    }
 
-        } else if (tagName.equals("input")) {
-            String inputType = ele.getAttribute("type");
-            scrollToView(by);
+    private void handleTextareaInput(WebElement element, String dataToInput) {
```
Comment (anandbagmar) [inline src/main/java/com/znsio/rpap/pages/BasePage.java:42]: What does "handle" do?

File: src/main/java/com/znsio/rpap/pages/BasePage.java (hunk 1 above)
Comment (raghavgarg1996) [reply src/main/java/com/znsio/rpap/pages/BasePage.java]: Updated method name

File: src/main/java/com/znsio/rpap/pages/BasePage.java (hunk 2 above)
Comment (raghavgarg1996) [reply src/main/java/com/znsio/rpap/pages/BasePage.java]: Updated method name

**Review by anandbagmar — APPROVED** [review]

**Pull request merged**